### PR TITLE
Fixes #22849 - update rubocop to 0.54

### PR DESCRIPTION
**Pull request by mmoll**
[ci skip]

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -142,28 +137,34 @@ Layout/SpaceInsidePercentLiteralDelimiters:
...
 
-# Offense count: 113
+# Offense count: 121
 Lint/AmbiguousRegexpLiteral:
   Enabled: false
 
-# Offense count: 89
+# Offense count: 1
+# Configuration parameters: AllowSafeAssignment.
+Lint/AssignmentInCondition:
```
Comment (mmoll) [inline .rubocop_todo.yml]: @bastilian could you have a look there? It's app/services/report_scanner/puppet_report_scanner.rb:5

File: bundler.d/development.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 group :development do
   gem 'maruku', '~> 0.7'
-  gem 'rubocop', '0.52.1'
+  gem 'rubocop', '~> 0.53.0'
```
Comment (tbrisker) [inline bundler.d/development.rb]: hound seems to still be on 0.52.1 (https://github.com/houndci/linters/blob/master/Gemfile.lock#L78), so it won't benefit from the new rules until they upgrade their version

File: .rubocop_todo.yml [hunk 3]
```diff
@@ -267,12 +269,16 @@ Naming/FileName:
...
 
-# Offense count: 20
+# Offense count: 23
 # Configuration parameters: Blacklist.
 # Blacklist: END, (?-mix:EO[A-Z]{1})
 Naming/HeredocDelimiterNaming:
   Enabled: false
 
+# Offense count: 19
+Naming/MemoizedInstanceVariableName:
```
Comment (tbrisker) [inline .rubocop_todo.yml]: should we add new issues in redmine for new cops?

Comment (mmoll) [reply .rubocop_todo.yml]: I would say only if one does open a PR for fixing.

File: .rubocop_todo.yml (hunk 1 above)
Comment (bastilian) [reply .rubocop_todo.yml]: Sorry for the delay. Here: https://github.com/theforeman/foreman/pull/5341

**Comment by mmoll** [comment]
https://github.com/houndci/linters/pull/184 got merged

**Review by tbrisker — DISMISSED** [review]
Thanks @mmoll !

**Comment by tbrisker** [comment]
Needs ack from @theforeman/packaging so GH permits merging.

**Comment by tbrisker** [comment]
@mmoll i merged the other rubocop pr, so this will need rebasing or regenerating the todo file

**Review by tbrisker — APPROVED** [review]

**Review by dLobatog — APPROVED** [review]
Thanks @mmoll !

**Pull request merged**